haskell/src/Stablecoin/Client/Parser.hs

```diff
@@ -300,11 +302,27 @@ clientArgsRawParser = Opt.subparser $
           (#help .! "Use the provided off-chain URI for metadata URI"))
 
         inplaceOptParser :: Opt.Parser ContractMetadataOptions
-        inplaceOptParser =
-          Opt.flag' OpCurrentContract
-            (Opt.long "contract-metadata-in-place"
-            <> Opt.help
-              "Embed the metadata for the contract in the contract's storage itself.")
+        inplaceOptParser = do
+          _ <-
+            Opt.flag' ()
+              (Opt.long "contract-metadata-in-place"
+              <> Opt.help
+                "Embed the metadata for the contract in the contract's storage itself.")
+          desc <- contractDescriptionParser
+          pure $ OpCurrentContract desc
+
+        remoteContractOptParser :: Opt.Parser ContractMetadataOptions
+        remoteContractOptParser = OpRemoteContract <$> contractDescriptionParser
+
+        contractDescriptionParser :: Opt.Parser (Maybe Text)
+        contractDescriptionParser =
+          optional $ mkCLOptionParser @Text
+            Nothing
+            (#name .! "description")
+            (#help .! "An optional description of the contract to add to the contract's metadata. \
```

Comment:
Same comment as before on `add`, although here is less important given the explanation following.

Could be rephrased in a different way as well.